src/dashboard/src/main/management/commands/update_elasticsearch_mappings.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+"""Update Elasticsearch mappings via Put Mapping.
+
+Updates the Elasticsearch mappings for the aips and aipfiles indices for
+Archivematica 1.12 to enable sorting on all fields displayed in the new
+Archival Storage DataTable.
+
+Execution example:
+
+./manage.py update_elasticsearch_mappings
+"""
+from __future__ import absolute_import, print_function
+
+import sys
+
+from django.conf import settings
+from elasticsearch import ElasticsearchException
+
+from main.management.commands import DashboardCommand
+import elasticSearchFunctions as es
+
+
+class Command(DashboardCommand):
+
+    help = __doc__
+
+    def handle(self, *args, **options):
+        # Check that AIPs index is enabled
+        if es.AIPS_INDEX not in settings.SEARCH_ENABLED:
+            self.error(
+                "The AIPs indexes are not enabled. Please, make sure to "
+                "set the *_SEARCH_ENABLED environment variables to `true` "
+                "to enable the AIPs and Transfers indexes, or to `aips` "
+                "to only enable the AIPs indexes."
+            )
+            sys.exit(1)
+
+        es.setup_reading_from_conf(settings)
+        es_client = es.get_client()
+
+        try:
+            es_client.info()
+        except ElasticsearchException:
+            print("Error: Elasticsearch may not be running.")
+            sys.exit(1)
+
+        # Update AIPs index mappings
+        es_client.indices.put_mapping(
+            index=es.AIPS_INDEX,
+            doc_type=es.DOC_TYPE,
+            body={
+                "properties": {
+                    "accessionids": {"type": "keyword"},
+                    "status": {"type": "keyword"},
+                    "file_count": {"type": "integer"},
+                }
+            },
+        )
+
+        # Update AIP files index mapping
+        es_client.indices.put_mapping(
+            index=es.AIP_FILES_INDEX,
+            doc_type=es.DOC_TYPE,
+            body={
+                "properties": {
+                    "accessionid": {"type": "keyword"},
+                    "status": {"type": "keyword"},
+                    "filePath": {
+                        "type": "text",
+                        "analyzer": "file_path_and_name",
+                        "fields": {"raw": {"type": "keyword"}},
```

Comment:
Is this new raw field populated in existing documents when the mapping is modified? If that's not the case, I think it's the only one that we could populate in this script based on the current value of the text field. You could use the [update by query API](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/docs-update-by-query.html) and a [Painless script](https://www.elastic.co/guide/en/elasticsearch/painless/6.8/painless-examples.html) to do so if it's needed.

Considering that we can't do the same for the other new fields, feel free to disregard if this is something outside the scope of this feature. Nevertheless, it would be nice to note this limitations in the Archivematica documentation.

Reply:
It looks like the new raw field is not updated, so good catch! I will look into populating that field in this script as I have time this week.

I'll also make sure that the limitations of the script are clear in the Archivematica documentation and on our radar for 1.12 client upgrades, so that everyone's aware of what this script does and doesn't do, and that full reindexing is necessary if one wants to fully populate these new fields for AIPs ingested prior to the upgrade.

Reply:
Thinking it through ... If we still have to document that limitation for all the other fields, it probably doesn't deserve the effort to fix it for this one. Totally up to you ;)

Reply:
@jraddaoui Hi Radda - coming back to this after a little while, but I went ahead and added the query by update to populate the `filePath.raw` subfield because some Archivematica users may not want or be able to fully reindex and this seems like a rather crucial field to be able to sort on. Thanks for the suggestion!

Would you mind taking a quick look when you have a minute to make sure the new updates look good? I have tested it locally and it worked well and quickly.